be/src/common/exception_stack_trace.h

```diff
@@ -0,0 +1,238 @@
+        }
+
+        // Resolve the source file name using DWARF:
+        if (dwfl != NULL) {
+            addr = (uintptr_t)(ip - 4);
+            [[maybe_unused]] Dwfl_Module* module = dwfl_addrmodule(dwfl, addr);
+            // Here we could also ask for the procedure name:
+            //const char* vProcedureName = dwfl_module_addrname(module, addr);
+            // Here we could also ask for the object file name:
+            //vDynObjectFileName = dwfl_module_info(module, NULL, NULL, NULL, NULL, NULL, NULL, NULL);
+            vDWARFObjLine = dwfl_getsrc(dwfl, addr);
+            if (vDWARFObjLine != NULL) {
+                vSourceFileName = dwfl_lineinfo(vDWARFObjLine, &addr, &vSourceFileLineNumber, NULL, NULL, NULL);
+                //fprintf(stderr, " %s:%d", strrchr(vSourceFileName, '/')+1, vSourceFileLineNumber);
+            }
+        }
+        if (dwfl == NULL || vDWARFObjLine == NULL || vSourceFileName == NULL) {
+            vSourceFileName = "???";
+            vSourceFileLineNumber = 0;
+        }
+
+        // Print the stack frame number:
+        fprintf(stderr, "#%2d:", ++n);
+
+        // Print the stack addresses:
+        fprintf(stderr, " 0x%016" PRIxPTR " sp=0x%016" PRIxPTR, static_cast<uintptr_t>(ip), static_cast<uintptr_t>(sp));
+
+        // Print the source file name:
+        fprintf(stderr, " %s:%d", vSourceFileName, vSourceFileLineNumber);
+
+        // Print the dynamic object file name (that is the library name).
+        // This is typically not interesting if we have the source file name.
+        //fprintf(stderr, " %s", vDynObjectFileName);
+
+        // Print the procedure name:
+        fprintf(stderr, " %s", vProcedureName);
+
+        // Print the procedure offset:
+        //fprintf(stderr, " + 0x%" PRIxPTR, static_cast<uintptr_t>(off));
+
+        // Print a newline to terminate the output:
+        fprintf(stderr, "\n");
+
+        // Stop the stack trace at the main method (there are some
+        // uninteresting higher level functions on the stack):
+        if (strcmp(vProcedureName, "main") == 0) {
+            break;
+        }
+
+        vUnwindStatus = unw_step(&vUnwindCursor);
+        if (vUnwindStatus < 0) {
+            fprintf(stderr, "libunwind_print_backtrace(): Error in unw_step: %d\n", vUnwindStatus);
+        }
+    }
+}
+
+void __wrap___cxa_throw(void* thrown_exception, void* infov, void (*dest)(void*)) {
+    std::type_info* info = (std::type_info*)infov;
+    auto query_id = starrocks::CurrentThread::current().query_id();
+    auto fragment_instance_id = starrocks::CurrentThread::current().fragment_instance_id();
+    fprintf(stderr, "query_id=%s, fragment_instance_id=%s throws exceptions:", starrocks::print_id(query_id).c_str(),
+            starrocks::print_id(fragment_instance_id).c_str());
+    // print the stack trace to stderr:
+    if (mExceptionStackTrace) {
+        print_exception_info(info);
+        libunwind_print_backtrace(1);
+    }
```

Comment:
Is it possible to throw an exception when calling these two functions, such as std::bac_alloc because of insufficient memory? is it necessary to add try catch here?

Reply:
if print_exception_info or libunwind_print_backtrace throw exceptions, then recurisively call __cta_throw(), the catch does not work. we may force these two functison not throwing exceptions.